src/context/DataContext.jsx

```diff
@@ -0,0 +1,23 @@
+import { createContext, useContext, useState } from 'react';
+
+const DataContext = createContext();
+
+export const useData = () => useContext(DataContext);
+
+export const DataProvider = ({ children }) => {
+  const [citiesData, setCitiesData] = useState([]);
+  const [isDataLoaded, setIsDataLoaded] = useState(false);
+
+  const updateCitiesData = data => {
+    setCitiesData(data);
+    setIsDataLoaded(true);
+  };
+
+  return (
+    <DataContext.Provider
+      value={{ citiesData, isDataLoaded, updateCitiesData }}
```

Comment:
wrap the value to `useMemo` to avoid unnecessary rerenders

Reply:
Yes, that's right. But I did it on purpose.
In our case, citiesData and isDataLoaded are simple states that don't change very often. citiesData is an array, and isDataLoaded is a boolean. Changing these states doesn't cause significant computational overhead.
updateCitiesData is a simple function that changes the state. It doesn't do any heavy lifting and doesn't depend on other states or props.
Components that use the context simply access the data and display it. There are no complex calculations that could cause extra renders.

Reply:
Anyway I'll do it, no problem.

Reply:
I've checked, in this case, yes, but It's anti-pattern

check the thread out
https://t.me/iamakulov_channel/594